src/common/src/cache.rs

```diff
@@ -769,22 +770,84 @@ impl<K: LruKey, T: LruValue> LruCache<K, T> {
 
 pub struct CleanCacheGuard<'a, K: LruKey + Clone + 'static, T: LruValue + 'static> {
     cache: &'a Arc<LruCache<K, T>>,
-    key: K,
+    key: Option<K>,
```

Comment:
This option is only for `make_success` right? Why don't we keep the original flag `success` ?

Reply:
> This option is only for `make_success` right? Why don't we keep the original flag `success` ?

We want to move `key` field out of this struct after `mark_success` is called